File: 프로그래머스/2/84512. 모음 사전/모음 사전.js
var str_arr = ["A", "E", "I", "O", "U"]
var array = [];

function solution(word) {
    var answer = 0;
    search("",0);
    for(var i = 0; i < array.length; i++){
      if(array[i] === word)
        break;
      else answer++;
    }
    return answer;
}

function search(word, num){
  array.push(word);
  if(num === 5)
    return;
  for(var i = 0; i < 5; i++){
    search(word+str_arr[i], num+1);
  }
}

File: 프로그래머스/2/84512. 모음 사전/모음 사전.py
def solution(word):
    answer = 0
    word_list = []
    words = "AEIOU"
    
    def search(cnt, w):
        if cnt == 5:
            return
        for i in range(len(words)):
                word_list.append(w + words[i])
                search(cnt + 1, w + words[i])
                
            
    search(0, "")
    
    return word_list.index(word) + 1